zstacklib/zstacklib/utils/qmp.py

```
@@ -84,3 +128,176 @@ def qmp_subcmd(qemu_version, s_cmd):
+    def command(self, cmd, **kwds):
+        """
+        Build and send a QMP command to the monitor, report errors if any
+        """
+        if kwds:
+            for k, v in kwds.items():
+                if "_" in k:
+                    kwds[k.replace("_", "-")] = v
+                    del kwds[k]
+        ret = self._cmd(cmd, kwds)
+        if "error" in ret:
+            raise Exception(ret['error']['desc'])
```

Ruff (0.14.4)

Create your own exception

(TRY002)